scality_manila_utils/smb_helper.py

```diff
@@ -0,0 +1,359 @@
+# Copyright (c) 2015 Scality
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+# implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Collection of functions for
+ - Addition and removal of exports
+ - Management of client permissions on export locations
+"""
+
+import errno
+import functools
+import io
+import json
+import logging
+import os
+import subprocess
+import sys
+import time
+
+try:
+    import ConfigParser as configparser
+except ImportError:
+    import configparser
+
+
+from scality_manila_utils import utils
+from scality_manila_utils.exceptions import (ClientExistsException,
+                                             ClientNotFoundException,
+                                             EnvironmentException,
+                                             ExportException,
+                                             ExportAlreadyExists,
+                                             ExportNotFoundException,
+                                             ExportHasGrantsException)
+
+log = logging.getLogger(__name__)
+if sys.version_info[0] == 2:
+    text_type = unicode
+else:
+    text_type = str
+
+
+# From http://prosseek.blogspot.fr/2012/10/
+# reading-ini-file-into-dictionary-in.html
+class MyIniParser(configparser.ConfigParser):
+    def as_dict(self):
+        d = dict(self._sections)
+        for k in d:
+            d[k] = dict(self._defaults, **d[k])
+            d[k].pop('__name__', None)
+        return d
+
+
+def _get_defined_exports():
+    """Retrieve all defined exports from the Samba registry."""
+
+    with utils.elevated_privileges():
+        process = subprocess.Popen(['net', 'conf', 'list'],
```

Comment:
If an OSError is raised, what should I do with it ? Not catching it would stop the program which seems to good thing to do. What do you have in mind ?